app/features/workout/WorkoutScreen.js

```diff
@@ -81,14 +81,25 @@ const createViewModels = (state, sets) => {
         }
 
         // rest footer
-        if (isInitialSet) {
+        if (isInitialSet && isLastSet) {
+            // new set, reset the end time
+            lastSetEndTime = isRemoved ? null : SetTimeCalculator.endTime(set);
+            array.push(createBottomBorder(set));
+        } else if (isInitialSet && !isLastSet && !isCollapsed) {
             // new set, reset the end time
             lastSetEndTime = isRemoved ? null : SetTimeCalculator.endTime(set);
+            array.push(createDeleteFooter(set));
```

Comment:
- [ ] nest the array.push(createDeleteFooter(set)); in the previous if statement